examples/asyncio_displayio_button.py

```diff
@@ -0,0 +1,312 @@
+    using make_transparent() and make_opaque() on it's palette.
+
+    :param displayio.Palette palette: The palette to change colors on for blinking
+    :param Interval interval: The Interval data object containing the interval length to use
+    :param int count: The number of times to repeat the blink. -1 for indefinite loop
+    :param ButtonState button_state: The ButtonState data object for the invert color button
+    """
+    while count < 0 or count > 0:
+
+        # if the color button is pressed
+        if button_state.state:
+            # if the color is still on default
+            if palette[0] == palette[1]:
+                # invert the color by subtracting from white
+                palette[0] = 0xFFFFFF - palette[0]
+
+        # if the color button is not pressed
+        else:
+            # set the color back to default
+            palette[0] = palette[1]
+
+        # hide the circle
+        palette.make_opaque(0)
+        # wait interval length
+        await asyncio.sleep(interval.value / 1000)  # Don't forget the await!
+
+        # show the circle
+        palette.make_transparent(0)
+        # wait interval length
+        await asyncio.sleep(interval.value / 1000)  # Don't forget the await!
+
+        # decrement count if it's positive
+        if count > 0:
+            count -= 1
+
+
+async def monitor_color_button(button, button_state):
+    """
+    monitor_color_button coroutine. Checks if the color button
+    is pressed, and updates the ButtonState data object as appropriate
+
+    :param Button button: The button to check for presses on
+    :param ButtonState button_state: ButtonState data object to set
+     the current value into
+    """
+    while True:
+        # get current touch data from overlay
+        p = ts.touch_point
+
+        # if there are any touch events
+        if p:
+            # if the color button is being touched
+            if button.contains(p):
+                # set selected to change button color
+                button.selected = True
+                # set button_state so other coroutines can access it
+                button_state.state = True
+
+            # the color button is not being touched
+            else:
+                # set selected to change button color back to default
+                button.selected = False  # if touch is dragged outside of button
+                # set button_state so other coroutines can access it.
+                button_state.state = False
+
+        # there are no touch events
+        else:
+            # if the color button is currently the pressed color
+            if button.selected:
+                # set selected back to false to change button back to default color
+                button.selected = False
+                # set button_state so other coroutines can access it
+                button_state.state = False
+
+        # give other tasks the opportunity to work
+        await asyncio.sleep(0)
```

Comment:
I think you would use `yield` rather than `await asyncio.sleep(0)`.

Reply:
I don't think I understand the difference do you know of a resource that explains? I adapted the code here from the examples in this guide: https://learn.adafruit.com/cooperative-multitasking-in-circuitpython-with-asyncio/communicating-between-tasks which uses `await asyncio.sleep(0)` inside of a similar coroutine that monitors physical buttons instead of virtual ones.

Reply:
Nope, don't use `yield`. This is correct as written.

Reply:
`asyncio.sleep(0)` is recommended as the canonical way to yield control back to the event loop. It is not the style in MicroPython, but it is the style recommended in several tutorials I have seen.